src/plugin.ts

```diff
@@ -0,0 +1,212 @@
+
+    this.listenToHighlighterEvents();
+  }
+
+  public destroy = (): void => {
+    this.unwatchQueryParams();
+    this.nodeHighlighter.destroy();
+    this.iFrameCommunicator?.destroy();
+  };
+
+  public updateConfiguration = (configuration: Partial<IPluginConfiguration>): void => {
+    this.configuration = { ...(this.configuration || defaultConfiguration), ...configuration };
+  };
+
+  private updateState = (state: Partial<IPluginState>): void => {
+    this.state = { ...this.state, ...state };
+  };
+
+  private initializeIFrameCommunication = (): void => {
+    if (!isInsideIFrame() || this.iFrameCommunicator) return;
+
+    this.iFrameCommunicator = new IFrameCommunicator();
+
+    const storage = createStorage<IPluginIFrameSettings>('kk:plugin:iframe-settings');
+    const stored = storage.get();
+    const highlighterEnabled = stored !== null ? stored?.highlighterEnabled : true;
+
+    if (highlighterEnabled) {
+      this.nodeHighlighter.enable();
+    }
+
+    this.iFrameCommunicator.addMessageListener(
+      IFrameMessageType.HighlightsStatus,
+      (data: IHighlightsStatusMessageData) => {
+        if (!data || !this.nodeHighlighter) return;
+
+        if (data.enabled) {
+          this.nodeHighlighter.enable();
+        } else {
+          this.nodeHighlighter.disable();
+        }
+
+        storage.set({
+          highlighterEnabled: data.enabled,
+        });
+      }
+    );
+
+    this.iFrameCommunicator.sendMessage(IFrameMessageType.Initialized, {
+      projectId: this.configuration.projectId,
+      languageCodename: this.configuration.languageCodename,
+      highlighterEnabled,
+    });
+  };
+
+  private watchQueryParams = (): void => {
+    if (isInsideIFrame()) return;
+
+    if (this.state.queryTimerId) {
+      clearTimeout(this.state.queryTimerId);
+    }
+
+    const currentPageUrl = new URL(document.URL);
+    const isPluginEnabled = currentPageUrl.search.includes(this.configuration.queryParam);
+    const hasChanged = this.nodeHighlighter.enabled !== isPluginEnabled;
+
+    if (hasChanged) {
+      if (isPluginEnabled) {
+        this.nodeHighlighter.enable();
+      } else {
+        this.nodeHighlighter.disable();
+      }
+    }
+
+    this.updateState({
+      queryTimerId: setTimeout(this.watchQueryParams, 1000),
+    });
+  };
+
+  private unwatchQueryParams = (): void => {
+    clearTimeout(this.state.queryTimerId);
+  };
+
+  private listenToHighlighterEvents = (): void => {
+    this.nodeHighlighter.addEventListener(NodeHighlighterEventType.ElementClicked, this.onElementClick);
+  };
+
+  private onElementClick = (elementClickData: Partial<IElementClickMessageData>): void => {
+    const data: Partial<IElementClickMessageData> = {
+      ...elementClickData,
+      projectId: elementClickData.projectId || this.configuration.projectId || undefined,
+      languageCodename: elementClickData.languageCodename || this.configuration.languageCodename || undefined,
+    };
+
+    let errorsCount = 0;
```

Comment:
If you use `const errors = []` and `errors.push(error)` you don't need to use `let` here